### ISSUE TITLE
GitHub App does not show preview results in Pulumi Bot comment when running on self-hosted runner in K8s

### ISSUE THREAD

**Issue by videmsky** [pulumi/console-requests#457]
When executing `pulumi preview` in the context of a PR, the "rich" preview output is not displayed when job is run on a self-hosted runner deployed in k8s via the Actions Runner Controller installed via Helm.

https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/deploying-runner-scale-sets-with-actions-runner-controller

## Expected behavior

![Image](https://github.com/user-attachments/assets/efbeb8af-2640-4148-8a79-06dcfd17bfa0)

## Current behavior
Preview is correct when viewed in GH Actions job output, but preview is not displayed as a PR comment.

## Steps to reproduce

When running a pulumi preview in the context of a PR, I get "rich" preview output when running:
(1) gh hosted runner with `ubuntu-latest`
(2) self-hosted gh runner on my ARM64 mac

I do not get a "rich" preview when running:
(3) self-hosted gh runner installed with Helm on K8s. (local Kind cluster)
 
Even in the same PR, I can get the "rich" preview to show with a trivial commit and switching between (1) & (2). Another commit but targeting (3) does not generate the preview.

https://github.com/videmsky/aws-ts-static-website/pull/2

## Context (Environment)
Pulumi Preview as a comment is helpful during PR code reviews.

## Affected feature

Github App

**Comment by gunzy83** [comment]
This is also happening when running on [Codebuild based GH Actions self-hosted runners](https://aws.amazon.com/blogs/devops/aws-codebuild-managed-self-hosted-github-action-runners/) running inside our VPC. In my case, the previews are being reported unreliably rather than never:

![Image](https://github.com/user-attachments/assets/030ec1fa-d3e4-42ef-97aa-4d3e5b714d75)

The checks were:

* 1 project test run (prettier, eslint, unit tests)
* 6 preview job runs
* 6 Codebuild runs (one for each preview jobs)
* 3 Preview checks returned by the Pulumi Github App (should be 6)

**Comment by gunzy83** [comment]
The preview messages for Github Actions running on Codebuild runners are completely broken now and are simply not appearing. I have turned on preview messages from the Pulumi Action so at least we have something for those stacks.

**Comment by aureq** [comment]
@gunzy83 When you say broken, could you share a screen capture with us please? What do you get exactly please?

**Comment by gunzy83** [comment]
> @gunzy83 When you say broken, could you share a screen capture with us please? What do you get exactly please?

I can't screen capture anything because it's the absence of the expected behaviour. The app no longer comments at all on the PR when we run on a self hosted (codebuild based) runner. There are changes on all stacks on these PRs so I would expect to see comments.

Edit: I'll grab another screenshot showing the codebuild statuses when I'm at the computer next but it's similar to my previous comment on this issue, just not intermittent now.

**Comment by gunzy83** [comment]
These are the codebuild statuses:

<img width="1956" height="820" alt="Image" src="https://github.com/user-attachments/assets/179eff66-bf1f-4069-b075-9cf4d6d7dba1" />

This is the workflow with the matching build:

<img width="1418" height="469" alt="Image" src="https://github.com/user-attachments/assets/ae33f4b8-54a0-434d-bbfd-148ca41214f8" />

The preview in the cloud showing releaseVersion changes:

<img width="2856" height="1562" alt="Image" src="https://github.com/user-attachments/assets/5a015b3a-7e2f-42ab-9d31-933f20baa7fe" />

The cloud seems to be aware of it running in codebuild but its not just codebuild, this is a Github runner: https://docs.aws.amazon.com/codebuild/latest/userguide/action-runner.html

**Comment by gunzy83** [comment]
I was testing out some workflow changes today and comments about deployments were being posted to my PR for a stack I was testing the workflows with. I have never run deployments in the context of a PR but was triggering on push to my branch to test so I was surprised to see them.

Previews are still MIA and the deployment comments are persistent (I guess that makes sense to track changes made).

**Comment by lukebward** [comment]
As of early January this year, I was not able to reproduce the preview issue.  Using a self-hosted runner installed with Helm on Kind I was able to get the full preview information.  

Out of curiosity, what are you GitHub app integrations set to?  You should be able to see these toggles 

<img width="666" height="264" alt="Image" src="https://github.com/user-attachments/assets/69403808-baa2-4e14-955b-94be1fa78f9d" />

**Comment by gunzy83** [comment]
I do not think this is fixed for all types of custom runners (in my case CodeBuild based Github Actions runners) and I think it might be when the platform detection finds something it ignores the other platform. I know this issue was specific to k8s even though the issue seemed to be affecting other runners. Should I open a new issue for CodeBuild based runners?

Edit: I just forced renovate to rebase a PR that has AMI upgrades. Two projects generated previews but only the one that ran on a hosted runner showed previews. The CodeBuild based runs generate no comments despite there being changes.
